File: app/src/main/java/com/capstone/bankit/ui/customview/ReceiptBottomSheetFragment.java
package com.capstone.bankit.ui.customview;

import android.app.Dialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.capstone.bankit.R;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

public class ReceiptBottomSheetFragment extends BottomSheetDialogFragment {
    private OnImagePickerClickListener listener;

    public interface OnImagePickerClickListener {
        void cameraClick();
        void galleryClick();
    }

    public void setOnButtonClickListener(OnImagePickerClickListener listener) {
        this.listener = listener;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.receipt_bottom_sheet_fragment, container, false);

        view.findViewById(R.id.btn_scan).setOnClickListener(v -> {
            if (listener != null) listener.cameraClick();
            dismiss();
        });

        view.findViewById(R.id.btn_gallery).setOnClickListener(v -> {
            if (listener != null) listener.galleryClick();
            dismiss();
        });

        return view;
    }
}

File: app/src/main/java/com/capstone/bankit/ui/customview/ReceiptBottomSheetFragment.kt
package com.capstone.bankit.ui.customview

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.capstone.bankit.R
import com.google.android.material.bottomsheet.BottomSheetDialogFragment

class ReceiptBottomSheetFragment : BottomSheetDialogFragment() {
    private var listener: OnImagePickerClickListener? = null

    interface OnImagePickerClickListener {
        fun cameraClick()
        fun galleryClick()
    }

    fun setOnButtonClickListener(listener: OnImagePickerClickListener?) {
        this.listener = listener
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View? {
        val view = inflater.inflate(R.layout.receipt_bottom_sheet_fragment, container, false)

        view.findViewById<View>(R.id.btn_scan).setOnClickListener { v: View? ->
            if (listener != null) listener!!.cameraClick()
            dismiss()
        }

        view.findViewById<View>(R.id.btn_gallery).setOnClickListener { v: View? ->
            if (listener != null) listener!!.galleryClick()
            dismiss()
        }

        return view
    }
}